06_HelloGraphicsQueue/main.cpp

```diff
@@ -2,80 +2,436 @@
+			secondImgBlit.srcOffsets[0] = { 0u, 0u, 0u };
+			secondImgBlit.srcOffsets[1] = { bigImgExtent.width, bigImgExtent.height, 1u };
+			secondImgBlit.dstSubresource = subresourceLayers;
+			secondImgBlit.dstOffsets[0] = { 0u, 0u, 0u };
+			secondImgBlit.dstOffsets[1] = { smallImgExtent.width, smallImgExtent.height, 1u };
+		}
 
-			core::smart_refctd_ptr<IGPUCommandBuffer> cmdbuf;
-			if (!m_device->createCommandBuffers(nullptr,IGPUCommandBuffer::EL_PRIMARY,1,&cmdbuf))
+		core::smart_refctd_ptr<IGPUCommandBuffer> cmdbuf;
+		{
+			core::bitflag<IGPUCommandPool::E_CREATE_FLAGS> flags = static_cast<IGPUCommandPool::E_CREATE_FLAGS>(IGPUCommandPool::E_CREATE_FLAGS::ECF_TRANSIENT_BIT);
+			smart_refctd_ptr<nbl::video::IGPUCommandPool> cmdpool = m_device->createCommandPool(queue->getFamilyIndex(), flags);
+			if (!m_device->createCommandBuffers(cmdpool.get(), IGPUCommandBuffer::EL_PRIMARY, 1, &cmdbuf))
 				return false;
+		}
+
+		IGPUQueue::SSubmitInfo submitInfo{};
+		submitInfo.commandBufferCount = 1;
+		submitInfo.commandBuffers = &cmdbuf.get();
+
+		IGPUCommandBuffer::SImageMemoryBarrier imgLayoutTransitionBarrier = constructDefaultInitializedImageBarrier();
+		imgLayoutTransitionBarrier.newLayout = asset::IImage::EL_TRANSFER_DST_OPTIMAL;
+		imgLayoutTransitionBarrier.barrier.dstAccessMask = E_ACCESS_FLAGS::EAF_MEMORY_WRITE_BIT;
+
+		IGPUCommandBuffer::SImageMemoryBarrier imgClearBarrier = constructDefaultInitializedImageBarrier();
+		imgClearBarrier.oldLayout = asset::IImage::EL_TRANSFER_DST_OPTIMAL;
+		imgClearBarrier.barrier.srcAccessMask = E_ACCESS_FLAGS::EAF_MEMORY_WRITE_BIT;
+
+		constexpr SClearColorValue red = { .float32{1.0f, 0.0f, 0.0f, 1.0f} };
+		constexpr SClearColorValue blue = { .float32{0.0f, 0.0f, 1.0f, 1.0f} };
+
+		smart_refctd_ptr<IGPUFence> done = m_device->createFence(IGPUFence::ECF_UNSIGNALED);
+
+		// Start recording.
+		cmdbuf->begin(IGPUCommandBuffer::EU_ONE_TIME_SUBMIT_BIT);
+
+		// In order to use images, we need to change their layout from asset::IImage::EL_UNDEFINED.
+		// Here it is done with use of barriers.
+		IGPUCommandBuffer::SImageMemoryBarrier bigImgLayoutTransitionBarrier = imgLayoutTransitionBarrier;
+		bigImgLayoutTransitionBarrier.image = bigImg;
+
+		IGPUCommandBuffer::SImageMemoryBarrier smallImgLayoutTransitionBarrier = imgLayoutTransitionBarrier;
+		smallImgLayoutTransitionBarrier.image = smallImg;
+
+		// Transit layouts of both images.
+		cmdbuf->pipelineBarrier(
+			E_PIPELINE_STAGE_FLAGS::EPSF_HOST_BIT | E_PIPELINE_STAGE_FLAGS::EPSF_ALL_COMMANDS_BIT,
+			E_PIPELINE_STAGE_FLAGS::EPSF_TRANSFER_BIT,
+			E_DEPENDENCY_FLAGS::EDF_NONE, 0u, nullptr, 0u, nullptr, 1u,
+			&bigImgLayoutTransitionBarrier
+		);
+
+		cmdbuf->pipelineBarrier(
+			E_PIPELINE_STAGE_FLAGS::EPSF_HOST_BIT | E_PIPELINE_STAGE_FLAGS::EPSF_ALL_COMMANDS_BIT,
+			E_PIPELINE_STAGE_FLAGS::EPSF_TRANSFER_BIT,
+			E_DEPENDENCY_FLAGS::EDF_NONE, 0u, nullptr, 0u, nullptr, 1u,
+			&smallImgLayoutTransitionBarrier
+		);
```

Comment:
you should combine both barriers into one, after all  the command takes a _count_ of buffer and image barriers